editor/src/components/canvas/canvas-strategies/strategies/set-padding-strategy.tsx

```diff
@@ -0,0 +1,215 @@
+import { MetadataUtils } from '../../../../core/model/element-metadata-utils'
+import { stripNulls } from '../../../../core/shared/array-utils'
+import { ElementInstanceMetadataMap } from '../../../../core/shared/element-template'
+import { CanvasVector } from '../../../../core/shared/math-utils'
+import { optionalMap } from '../../../../core/shared/optional-utils'
+import { ElementPath } from '../../../../core/shared/project-file-types'
+import { assertNever } from '../../../../core/shared/utils'
+import { ParsedCSSProperties } from '../../../inspector/common/css-utils'
+import { stylePropPathMappingFn } from '../../../inspector/common/property-path-hooks'
+import { CSSCursor, EdgePiece } from '../../canvas-types'
+import { deleteProperties } from '../../commands/delete-properties-command'
+import { setCursorCommand } from '../../commands/set-cursor-command'
+import { setElementsToRerenderCommand } from '../../commands/set-elements-to-rerender-command'
+import { setProperty } from '../../commands/set-property-command'
+import { updateHighlightedViews } from '../../commands/update-highlighted-views-command'
+import { isZeroSizedElement } from '../../controls/outline-utils'
+import { PaddingResizeControl } from '../../controls/select-mode/padding-resize-control'
+import {
+  PaddingResizeStepIndicator,
+  PaddingResizeStepIndicatorProps,
+} from '../../controls/select-mode/padding-resize-step-indicator'
+import {
+  deltaFromEdge,
+  offsetPaddingByEdge,
+  paddingForEdge,
+  paddingToPaddingString,
+  simplePaddingFromMetadata,
+} from '../../padding-utils'
+import { CanvasStrategyFactory } from '../canvas-strategies'
+import {
+  controlWithProps,
+  emptyStrategyApplicationResult,
+  getTargetPathsFromInteractionTarget,
+  InteractionCanvasState,
+  strategyApplicationResult,
+} from '../canvas-strategy-types'
+import { InteractionSession } from '../interaction-state'
+import { getDragTargets, getMultiselectBounds } from './shared-move-strategies-helpers'
+
+const StylePaddingProp = stylePropPathMappingFn('padding', ['style'])
+const IndividualPaddingProps: Array<keyof ParsedCSSProperties> = [
+  'paddingTop',
+  'paddingBottom',
+  'paddingLeft',
+  'paddingRight',
+]
+
+export const setPaddingStrategy: CanvasStrategyFactory = (
+  canvasState,
+  interactionSession,
+  customStrategyState,
+) => {
+  const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
+  if (selectedElements.length === 0) {
```

Comment:
We should filter here for `selectedElements.length !== 1`, as otherwise we show padding controls when we have multiple elements selected (but just don't act when they are being dragged)

Reply:
great point! updated here and in the rest of the strategy as well